File: src/main/java/org/shark/example/datasource/mongo/pojo/ExampleDo.java
package org.shark.example.datasource.mongo.pojo;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;
import java.util.Map;

@Data
public class ExampleDo {
    @Id
    private String id;
    private Map<String, String> map;
    private List<String> list;
}

File: src/main/kotlin/org/shark/example/datasource/mongo/pojo/ExampleDo.kt
package org.shark.example.datasource.mongo.pojo

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class ExampleDo(
    @Id
    var id: String? = null,
    var map: Map<String, String>? = null,
    var list: List<String>? = null
)
